### PR TITLE
Feature user info

### PR DESCRIPTION
**Pull request by hald01**
delete all test file
add user info function

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,5 +13,11 @@ def User.digest(string)
         cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
         BCrypt::Password.create(string, cost: cost)
     end
+
+    validates :age, presence: true, length: { maximum: 3 }, allow_nil: true
+
+    validates :adress, presence: true, length: { maximum: 100 }, allow_nil: true
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/models/user.rb:19]: i will fix it  now

File: app/views/users/_form_edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <%= f.email_field :email, class: 'form-control' %>
+
+    <%= f.label :password %>
+    <%= f.password_field :password, class: 'form-control' %>
+
+    <%= f.label :password_confirmation, "Confirmation" %>
+    <%= f.password_field :password_confirmation, class: 'form-control' %>
+
+    <%= f.submit yield(:button_text), class: "btn btn-primary" %>
+<% end %>
```
Comment (Thole-Myfave) [inline app/views/users/_form_edit.html.erb:26]: add break line

Comment (Thole-Myfave) [reply app/views/users/_form_edit.html.erb:26]: apply the same way for another place

File: app/views/users/_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<li><pre>User <%= user.id %></pre>
+    <ul>
```
Comment (Thole-Myfave) [inline app/views/users/_user.html.erb:2]: Pls use 2 indent

File: app/models/user.rb [hunk 4]
```diff
@@ -13,5 +13,11 @@ def User.digest(string)
         cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST : BCrypt::Engine.cost
         BCrypt::Password.create(string, cost: cost)
     end
+
+    validates :age, presence: true, length: { maximum: 3 }, allow_nil: true
+
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/models/user.rb:18]: I want it to be easier to watch when i check code

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -2,7 +2,9 @@ module SessionsHelper
     def log_in(user)
         session[:user_id] = user.id
     end
-    # Returns the current logged-in user (if any).
+    def current_user?(user)
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/helpers/sessions_helper.rb:5]: i will fix it

File: db/seeds.rb [hunk 6]
```diff
@@ -5,3 +5,26 @@
...
+             password:              "852963",
+             password_confirmation: "852963",
+             age:   "18",
+             adress:"10.Ngo Gia Tu - Hai Chau - Da Nang",
+             phone: "19001000",
+             admin: true
+             )
+#
+for a in 2...10 do
+User.create(   name: "User 00#{a}",
```
Comment (Thole-Myfave) [inline db/seeds.rb:19]: indent and no need space before `name`

File: db/seeds.rb [hunk 7]
```diff
@@ -5,3 +5,26 @@
...
+            email: "test00#{a}@gmail.com",
+            password: "123456",
+            password_confirmation: "123456")    
+end
+#
+for a in 10...100 do
```
Comment (Thole-Myfave) [inline db/seeds.rb:25]: can we use another loop for that ?

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -1,9 +1,16 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user,  only: [:show, :edit, :update, :index]
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/controllers/users_controller.rb:2]: i will add :excpet

File: app/views/users/_form_edit.html.erb (hunk 2 above)
Comment (hald01) [reply app/views/users/_form_edit.html.erb:26]: i will fix it

File: app/views/users/_user.html.erb (hunk 3 above)
Comment (hald01) [reply app/views/users/_user.html.erb:2]: i will fix it

File: db/seeds.rb (hunk 6 above)
Comment (hald01) [reply db/seeds.rb:19]: i will fix it

File: db/seeds.rb (hunk 7 above)
Comment (hald01) [reply db/seeds.rb:25]: i will replace that

Comment (PhanMinhHuy) [reply db/seeds.rb:25]: Can you write down 3 other ways to do this? more is good =))

File: app/models/user.rb [hunk 9]
```diff
@@ -13,5 +13,11 @@ def User.digest(string)
...
+    validates :adress, presence: true, length: { maximum: 100 }, allow_nil: true
+
+    validates :phone, presence: true, length: { maximum: 12 }, length: { minimum: 6 }, allow_nil: true
```
Comment (PhanMinhHuy) [inline app/models/user.rb:21]:
```suggestion
    validates :phone, presence: true, length: { maximum: 12, minimum: 6 }, allow_nil: true
```

File: app/views/users/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<% provide(:title, 'All users') %>
+<h1>All users</h1>
+
+<%= will_paginate %>
+
+<ul class="users">
+  <% @users.each do |user| %>
+      <%= render @users %>
+  <% end %>
```
Comment (PhanMinhHuy) [inline app/views/users/index.html.erb:9]:
```suggestion
  <%= render @users %>
```

File: db/seeds.rb (hunk 7 above)
Comment (hald01) [reply db/seeds.rb:25]: i will do it

File: app/controllers/users_controller.rb [hunk 11]
```diff
@@ -1,9 +1,16 @@
...
+  before_action :correct_user,    only: [:show, :edit, :update, ]
+  before_action :admin_user,      only: :index
+
+  def new
+    @user = User.new
+  end
   def show
     @user = User.find(params[:id])
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/controllers/users_controller.rb:10]: i will do it

File: app/models/user.rb (hunk 9 above)
Comment (hald01) [reply app/models/user.rb:21]: i will try it

File: app/views/users/index.html.erb (hunk 10 above)
Comment (hald01) [reply app/views/users/index.html.erb:9]: i will try it

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -15,8 +22,33 @@ def create
...
   private
   def user_params
-    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    params.require(:user).permit(:name, :email, :password, :password_confirmation, :age, :adress, :phone)
+  end
+  def logged_in_user
+    unless logged_in?
+      flash[:danger] = "Please log in."
+      redirect_to signin_url
+    end
```
[the remark this answers is not in the document]
Comment (hald01) [reply app/controllers/users_controller.rb:45]: i will try it

**Pull request merged**